test/powershell/engine/Api/TypeInference.Tests.ps1

```diff
@@ -302,6 +301,70 @@ Describe "Type inference Tests" -tags "CI" {
         }
     }
 
+    It "Infers type from foreach-object with membername" {
+        $res = [AstTypeInference]::InferTypeOf( { Get-ChildItem | ForEach-Object -MemberName Directory }.Ast)
+        $res.Count | Should Be 1
+        $res.Name | Should Be "System.IO.DirectoryInfo"
+    }
+
+    It 'Infers typeof Foreach-Object -Member when Member is Property' {
+        $ast = {Get-Process | Foreach-Object -Member FileVersion}.Ast
+        $typeNames = [AstTypeInference]::InferTypeof($ast, [TypeInferenceRuntimePermissions]::AllowSafeEval)
+        $typeNames.Count | Should be 1
+        $typeNames[0] | Should be 'System.String'
+    }
+
+    It 'Infers typeof Foreach-Object -Member when member is ScriptProperty' {
+        $ast = {Get-Process | Foreach-Object -Member Description}.Ast
+        $typeNames = [AstTypeInference]::InferTypeof($ast, [TypeInferenceRuntimePermissions]::AllowSafeEval)
+        $typeNames.Count | Should be 1
+        $typeNames[0] | Should be 'System.String'
+    }
+
+    It 'Infers typeof Foreach-Object -Member when Member is Alias' {
+        $ast = {Get-Process | Foreach-Object -Member Handles}.Ast
+        $typeNames = [AstTypeInference]::InferTypeof($ast, [TypeInferenceRuntimePermissions]::AllowSafeEval)
+        $typeNames.Count | Should be 1
+        $typeNames[0] | Should be 'System.Int32'
+    }
+
+    It 'Infers typeof Foreach-Object -Member when using dependent scriptproperties' {
+        class XSP {
+            [string] $Value
+            X() {
+                $this.Value = "TheValue"
+            }
+        }
+        class YSP {
+            [XSP] $X
+            YSP() {$this.X = [XSP]::new()}
+        }
+        Update-TypeData -TypeName XSP -MemberName TheValue -MemberType ScriptProperty -Value { return $this.Value } -Force
+        Update-TypeData -TypeName YSP -MemberName XVal -MemberType ScriptProperty -Value {return $this.X } -Force
+
+        $ast = {[YSP]::new() | Foreach-Object -MemberName XVal | ForEach-Object -MemberName TheValue}.Ast
+        $typeNames = [AstTypeInference]::InferTypeof($ast, [TypeInferenceRuntimePermissions]::AllowSafeEval)
+        $typeNames.Count | Should be 1
+        $typeNames[0] | Should be 'System.String'
+    }
+
+    # Not yet implementet
+    It 'Infers typeof Foreach-Object -Member when using dependent scriptproperties on pscustomobject' -Skip {
```

Comment:
Spelling, and to be honest, I don't think we want tests for features that might never get implemented.